db/transaction/transaction_test.go

```diff
@@ -50,4 +57,131 @@ var _ = Describe("Transaction", func() {
 			Expect(tx.GetIsolationLevel(netSchema, "update")).To(Equal(tx.Serializable))
 		})
 	})
+
+	Describe("CachedTransactionTest", func() {
+		var transx *mocks.MockTransaction
+		var cachedTx *sql.CachedTransaction
+		var count int
+		var mockCtrl *gomock.Controller
+
+		BeforeEach(func() {
+			mockCtrl = gomock.NewController(GinkgoT())
+			transx = mocks.NewMockTransaction(mockCtrl)
+			cachedTx = &sql.CachedTransaction{transx, nil}
+			cachedTx.ClearCache()
+
+			manager := schema.GetManager()
+			basePath := "../../tests/test_abstract_schema.yaml"
+			Expect(manager.LoadSchemaFromFile(basePath)).To(Succeed())
+
+			schemaPath := "../../tests/test_schema.yaml"
+			Expect(manager.LoadSchemaFromFile(schemaPath)).To(Succeed())
+			netSchema, _ = manager.Schema("network")
+
+			count = 0
+			transx.EXPECT().ListContext(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(context.Context, *schema.Schema, transaction.Filter, *transaction.ViewOptions, *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+				count++
+				return []*schema.Resource{}, 0, nil
+			}).AnyTimes()
+			transx.EXPECT().CreateContext(gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().UpdateContext(gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().DeleteContext(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+			transx.EXPECT().QueryContext(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+		})
+		AfterEach(func(){
+			mockCtrl.Finish()
+		})
+
+		It("First list not cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+		})
+
+		It("List calls get cached", func() {
+			Expect(count).To(Equal(0))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.List(netSchema, tx.Filter{}, nil, nil)
+			Expect(count).To(Equal(1))
+			cachedTx.LockList(netSchema, tx.Filter{}, nil, nil, schema.LockRelatedResources)
```

Comment:
First call to Lock* methods should never be cached - gohan relies on MySQL locks for synchronization